### ISSUE TITLE
src/unix/stream.c:1560

### ISSUE THREAD

**Issue by lexansoft** [xmrig/xmrig#1183]
Sep 20 15:10:35 pe-1 xmrig[89982]: xmrig: src/unix/stream.c:1560: uv_read_start: Assertion `stream->type == UV_TCP || stream->type == UV_NAMED_PIPE || stream->type == UV_TTY' failed.
Sep 20 15:10:35 pe-1 systemd[1]: xmrig.service: Main process exited, code=dumped, status=6/ABRT
Sep 20 15:10:35 pe-1 systemd[1]: xmrig.service: Failed with result 'core-dump'.

**Comment by xmrig** [comment]
Please provide full details, how I can reproduce the issue.
Thank you.

**Comment by lexansoft** [comment]
When I try to start xmrig as a service on ubuntu.

The service file:

[Unit]
Description=XMRIG Miner

Wants=network.target
After=syslog.target network-online.target

[Service]
User=alexn
Type=simple
ExecStart=/home/alexn/xmrig/build/xmrig -c /home/alexn/xmrig-config.json
Restart=on-failure
RestartSec=10
KillMode=process

[Install]
WantedBy=multi-user.target
~

**Comment by xmrig** [comment]
I confirm the bug, for workaround you can change 2 lines:

```
Type=forking
ExecStart=/home/alexn/xmrig/build/xmrig -c /home/alexn/xmrig-config.json -B
```

**Comment by lexansoft** [comment]
I do not think this is right. When started with the 'background' option, the process spawn the background sub-process and terminates. After that the main service PID does not exist anymore, so you cannot stop the service with: 

service xmrig stop

I think the right solution should be just staying in the main process, but if there is no terminal assigned, just continue without reading from the stdio.

**Comment by xmrig** [comment]
Fixed in dev and evo branches, option `StandardOutput=null` useful to prevent spamming to syslog, dedicated `syslog` option do job better (if need).

Also I found another bug in forking mode miner crashed on exit, I will check it tomorrow.
Thank you.

**Comment by lexansoft** [comment]
Great. Thank you.

Alexandre Naverniouk

**Comment by xmrig** [comment]
Fixed in v3.2.0 and v4.2.0.
